### ISSUE TITLE
Multiple bind from exchange to queue

### ISSUE THREAD

**Issue by ZinkevychSergii** [golevelup/nestjs#79]
Hi. Is't possible to bind queue to multiple routing keys from exchange.

example:
Have exchange `someExchange` with type topic
Have queue `someQueue`, and i would like `someExchange` will send messages to `someQueue` with routing keys ['some.#', '*.another'].

Code below bind only latest routing key at the array

```
export const RabbitSubscribeAndBindQueue = ({name: queue, binds}: IRabbitQueueConfig) => (
  target: any,
  propertyKey: string,
  descriptor: PropertyDescriptor,
) => {
  binds.forEach(({exchange, routingKey}) => {
    RabbitSubscribe({
      exchange,
      queue,
      routingKey,
    })(target, propertyKey, descriptor);
  });
};
```

**Comment by WonderPanda** [comment]
Hey @ZinkevychSergii I think this is a reasonable feature. Until it's implemented, have you tried adding multiple decorators to the same service method? I believe that will also get you what you want

**Comment by ZinkevychSergii** [comment]
@WonderPanda have tried, basically it's the same as at my snippet above, so unfortunately doesn't works. 
the core issue, when you use @RabbitSubscribe, under the hood SetMetadata(KEY) used. thing of metadata as hash map, when use SetMetada twice, the latest call will override previous one.

should be good enough just change routingKey to be an array and iterate here https://github.com/golevelup/nestjs/blob/master/packages/rabbitmq/src/amqp/connection.ts#L109
and change interfaces along the way.

who is interested in this feature until package doesn't support it can use my solution build on top of 
nest-plus/rabbitmq


```
export interface IRabbitQueueBindConfig {
  exchange: string;
  routingKey: string;
}

export interface IRabbitQueueConfig {
  name: string;
  binds: IRabbitQueueBindConfig[];
}
export const RabbitSubscribeByConfig = (config: IRabbitQueueConfig) => (
  target: any,
  propertyKey: string,
  descriptor: PropertyDescriptor,
) => {
  SetMetadata(RABBIT_MULTIPLE_ROUTING_HANDLER, config)(target, propertyKey, descriptor);
};
@Module({
  imports: [AmqpConnection, DiscoveryModule],
  providers: rabbitExchangeProviders,
  exports: [AmqpConnection, ...rabbitExchangeProviders],
})
export class RabbitModule implements OnModuleInit {
  constructor(
    @Inject('AmqpConnection') private readonly amqpConnection,
    private readonly discover: DiscoveryService,
  ) {}
  async onModuleInit() {
    const data = [
      ...(await this.discover.providerMethodsWithMetaAtKey<IRabbitQueueConfig>(
        RABBIT_MULTIPLE_ROUTING_HANDLER,
      )),
      ...(await this.discover.controllerMethodsWithMetaAtKey<IRabbitQueueConfig>(
        RABBIT_MULTIPLE_ROUTING_HANDLER,
      )),
    ];
    data.forEach(({ discoveredMethod, meta }) => {
      const handler = discoveredMethod.handler.bind(discoveredMethod.parentClass.instance);
      meta.binds.forEach(bind =>
        this.amqpConnection.createSubscriber(handler, {
          ...bind,
          queue: meta.name,
        }),
      );
    });
  }
}
```

**Comment by WonderPanda** [comment]
I think this feature makes sense for the library. I'll take a look at getting it integrated tonight

**Comment by WonderPanda** [comment]
@ZinkevychSergii Just published `@golevelup/nestjs-rabbitmq@1.8.0` which supports this functionality.  It's important not to call `createSubscriber` multiple times because that creates multiple internal queues instead of just routing messages from multiple exchange locations to the same handler. You can easily update your code because `routingKey` property now supports `string | string[]` so you can do:
```
@RabbitSubscribe({
    exchange,
    routingKey: ['routingKey1', 'routingKey2'],
    queue: 'subscribeQueue',
  })
```
